src/expr/src/relation/mod.rs

```diff
@@ -1807,10 +1807,12 @@ impl MirRelationExpr {
     /// Leaves `self` in an unusable state, so this should only be used if `self` is about to be
     /// dropped or otherwise overwritten.
     pub fn destroy_carefully(&mut self) {
-        #[allow(deprecated)] // Having `maybe_grow` and no limit is the point here
-        self.visit_mut_post_nolimit(&mut |e| {
-            e.take_dangerous();
-        });
+        let mut todo = vec![self.take_dangerous()];
+        while let Some(mut expr) = todo.pop() {
+            for child in expr.children_mut() {
+                todo.push(child.take_dangerous());
+            }
+        }
```

Comment:
This is a non-trivial change, though it is meant to do the same thing just more easily.